querybook/server/lib/scheduled_datadoc/notification.py

```diff
@@ -58,18 +60,35 @@ def _get_datadoc_notification_params(
     )
 
 
+class User:
+    def __init__(self):
+        self.email = None
+        self.username = None
+
+
+def generate_user(with_str, value):
+    new_user = User()
+    if with_str == "email":
+        new_user.email = value
+    elif with_str == "slack":
+        new_user.username = value
+
+    return new_user
+
+
 @with_session
 def _send_datadoc_notification(
-    user_id: int,
+    notify_to: List[Dict],
     notify_with: str,
     notification_params: Dict,
     session=None,
 ):
-    user = User.get(id=user_id, session=session)
-    notify_user(
-        user=user,
-        template_name="datadoc_completion_notification",
-        template_params=notification_params,
-        notifier_name=notify_with,
-        session=session,
-    )
+    for recepient in notify_to:
+        user = user = generate_user(notify_with, recepient)
+        notify_user(
```

Comment:
as the `recipient` (also a typo above) doesn't have to be a querybook user, e.g. slack channel, so it wont make sense to keep using `notify_user` with a `user` parameter passed in. we'd think about update this function accordingly to support like slack channels.

And eventually this `notify_user` will call the `BaseNotifier.notify`, which only accepts a user as parameter, this may also need to be updated.

also did you test the slack channel notification?  the `SlackNotifier` only supports for username today, you may also need to update it as well.